test/evergreen/tcmalloc_install_or_build.sh

```diff
@@ -0,0 +1,94 @@
+#!/bin/bash
+#
+# Make a tcmalloc shared library avilable for preloading into the environment.
+
+set -u
+
+if [[ $# -ne 1 ]]; then
+    echo "Usage: $0 build-variant"
+    exit 1
+fi
+
+build_variant=$1
+
+# This is tcmalloc upstream revision 093ba93 source patched by the SERVER team.
+# https://github.com/mongodb-forks/tcmalloc/releases/tag/mongo-SERVER-85737
+PATCHED_SRC=mongo-SERVER-85737
+PATCHED_TGZ="${PATCHED_SRC}.tar.gz"
+PATCHED_TGZ_URL="https://github.com/mongodb-forks/tcmalloc/archive/refs/tags/${PATCHED_TGZ}"
+PATCHED_SRC_DIR="tcmalloc-${PATCHED_SRC}"
+
+# Sensitive variables retrieved through Evergreen so they won't be exposed
+# in source control.
+#
+# Export the AWS_* variables so that they are visible for the 'aws' cli.
+S3_URL=${s3_bucket_tcmalloc}
+export AWS_ACCESS_KEY_ID=${s3_access_key}
+export AWS_SECRET_ACCESS_KEY=${s3_secret_key}
+
+# Directory into which shared object will be installed.
+install_dir=$PWD
+tcmalloc_dir=TCMALLOC_LIB
+tcmalloc_so_dir=${install_dir}/${tcmalloc_dir}
+
+# Location in AWS S3 for prebuilt tcmalloc binaries.
+PREBUILT_TGZ="tcmalloc-${PATCHED_SRC}-${build_variant}.tgz"
+PREBUILT_URL="${s3_bucket_tcmalloc}/build/wt_prebuilt_tcmalloc/${PATCHED_SRC}/${PREBUILT_TGZ}"
+
+# Without the --quiet option 'aws s3 cp' will print out an error message if the file is
+# NOT present. This is an expected error case that is handled later in this script, but
+# emitting messages with "Error" in them, into the Evergreen log will cause confusion.
+echo "Attempting to download prebuilt tcmalloc: ${PREBUILT_URL}"
+aws s3 cp --quiet ${PREBUILT_URL} ${PREBUILT_TGZ}
+aws_ret=$?
+if [[ $aws_ret -eq 0 ]]; then
+    tar zxf $PREBUILT_TGZ
+    exit $?
+elif [[ $aws_ret -ne 1 ]]; then
+    echo "ERROR aws s3 download failed with code ${aws_ret}"
+    exit $aws_ret
+fi
+
+# Downloading a prebuilt copy failed. Assume it is because it doesn't exist for this build
+# variant. Attempt to create a shared object for this build variant and upload it.
+# From this point onward: any and all errors are fatal, and will cause the build to FAIL.
+set -e
```

Comment:
Would this line overwrite the `set -u` line above? Do we want to have both options set from here?

Reply:
Don't want to set it earlier, as the s3 download may fail, and we want to handle that if the status is 1 and fail for any other non-status.

Reply:
My actual question is why we don't use `set -eu` here. It's not a big deal, just hope to call out in case it's an undesired missing (guess the `set -u` on line #5 was a later-add).

Reply:
I think you are right @ershov. Thanks for pointing it out.